cloud/amazon/ec2_lc_find.py

```diff
@@ -0,0 +1,226 @@
+    returned: when Launch Configuration was found
+    type: string
+    sample: mykey
+keyname:
+    description: Launch Configuration security groups
+    returned: when Launch Configuration was found
+    type: list
+    sample: []
+kernel_id:
+    description: Launch Configuration kernel to use
+    returned: when Launch Configuration was found
+    type: string
+    sample: ''
+ram_disk_id:
+    description: Launch Configuration ram disk property
+    returned: when Launch Configuration was found
+    type: string
+    sample: ''
+associate_public_address:
+    description: Assign public address or not
+    returned: when Launch Configuration was found
+    type: boolean
+    sample: True
+...
+'''
+
+
+def find_launch_configs(client, module):
+    name_regex = module.params.get('name_regex')
+    sort_order = module.params.get('sort_order')
+    offset = module.params.get('limit')
+
+    paginator = client.get_paginator('describe_launch_configurations')
+
+    response_iterator = paginator.paginate(
+        PaginationConfig={
+            'MaxItems': 1000,
+            'PageSize': 100
+        }
+    )
+
+    for response in response_iterator:
+        response['LaunchConfigurations'] = filter(lambda lc: re.compile(name_regex).match(lc['LaunchConfigurationName']),
+                                                  response['LaunchConfigurations'])
+
+    results = []
+    for lc in response['LaunchConfigurations']:
+        data = {
+            'name': lc['LaunchConfigurationName'],
+            'arn': lc['LaunchConfigurationARN'],
+            'created_time': lc['CreatedTime'],
+            'user_data': lc['UserData'],
+            'instance_type': lc['InstanceType'],
+            'image_id': lc['ImageId'],
+            'ebs_optimized': lc['EbsOptimized'],
+            'instance_monitoring': lc['InstanceMonitoring'],
+            'classic_link_vpc_security_groups': lc['ClassicLinkVPCSecurityGroups'],
+            'block_device_mappings': lc['BlockDeviceMappings'],
+            'keyname': lc['KeyName'],
+            'security_groups': lc['SecurityGroups'],
+            'kernel_id': lc['KernelId'],
+            'ram_disk_id': lc['RamdiskId'],
+            'associate_public_address': lc['AssociatePublicIpAddress'],
+        }
+
+        results.append(data)
+
+    results.sort(key=lambda e: e['name'], reverse=(sort_order == 'descending'))
+
+    try:
+        if offset:
+            results = results[:int(offset)]
+    except TypeError:
+        module.fail_json(msg="Please supply numeric values for offset")
+    module.exit_json(changed=False, results=results)
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        region=dict(required=True, aliases=['aws_region', 'ec2_region']),
+        name_regex=dict(required=True),
+        sort_order=dict(required=False, default='ascending', choices=['ascending', 'descending']),
+        limit=dict(required=False),
```

Comment:
not necessary, but did you know you can avoid the type error checking on this later if you just add type='int' to the argspec dict for this arg?